pkgs/development/interpreters/emilua/setup-hook.sh

```diff
@@ -0,0 +1,8 @@
+addEmiluaLibPath () {
+    if test -d "$1/lib/emilua-0.6"
+    then
+        export EMILUA_PATH="${EMILUA_PATH-}${EMILUA_PATH:+:}$1/lib/emilua-0.6"
```

Comment:
Can you unhardcode this 0.6 version and use the emilua derivation version instead so when emilua changes version things don't break?

Reply:
This is the plugins folder, the software's structure requires that the folder does not contain the entire version name, so I can't use something like /lib/@version@. Maybe creating a symlink? But wouldn't that make the code more bureaucratic and unnecessary? 
Also, once a new version is released, can't I simply change the script to the new version nomenclature?

Reply:
Or maybe the script could be changed by consequence because it gets the version from the emilua version

Reply:
> Also, once a new version is released, can't I simply change the script to the new version nomenclature?

There is a lemma we learn when reading [Unix Power Tools](https://www.oreilly.com/library/view/unix-power-tools/0596003307/): computers are made to automate boring tasks.

And updating a version reference by hand is a boring task.

My suggestion is to change this to something like

```suggestion
        export EMILUA_PATH="${EMILUA_PATH-}${EMILUA_PATH:+:}$1/lib/emilua-@version@"
```

and use the `substituteAll` Nixpkgs functionality.

You can find examples of it around Nixpkgs, E.G.  udisks2.

Reply:
I just realized that I don't need the setupHook. I was only using it to set the path for looking for the plugins. The wrapProgram already does this. So, I changed it to an unhardcoded value. I tested it here, and it works perfectly. 
I was basing it on [this ](https://github.com/NixOS/nixpkgs/blob/f2c670a6089987d6ec564647444c74d43ab8f509/pkgs/development/libraries/gstreamer/core/default.nix)package.